support/pal/src/windows.rs

```diff
@@ -195,7 +195,7 @@ impl OwnedSocketExt for OwnedSocket {
             }
             // In theory SIO_SOCKET_TRANSFER_END could have changed `socket`, so
             // forget the handle and use the socket instead.
-            handle.into_raw_handle();
+            std::mem::forget(handle);
```

Comment:
Hmm. I don't love this lint, because now you have to be really careful to make sure `handle` isn't a type that also contains some heap allocation or whatever.

What was the lint? Can we do something like `let _gone = handle.into_raw_handle()` instead?

Reply:
The lint was unused must_use on the return value of into_raw_handle. Super specific. We could do something like that, but given the comment above this felt better.